File: python/inflence-on-js/examples/listparam_generator.js
function listParam(csvStr) {
  return {
    [Symbol.iterator]: function*() {
      var position = 0;
      var commaPosition = csvStr.indexOf(",", position);
      while (commaPosition != -1) {
        yield csvStr.slice(position, commaPosition);
        position = commaPosition + 1
        commaPosition = csvStr.indexOf(",", position);
      }
      yield csvStr.slice(position);
    },
    toString: function() {
      return csvStr;
    }
  }
}

var params = listParam("one,two,three");
console.log("My list param is " + params + ".");
for (var param of params)
    console.log("One of its params is " + param + ".")

File: python/inflence-on-js/examples/listparam_generator.py
class ListParam:

    def __init__(self, csv_str):
        self.csv_str = csv_str

    def __str__(self):
        return self.csv_str

    def __iter__(self):
        position = 0
        comma_position = self.csv_str.find(",", position)
        while comma_position != -1:
            yield self.csv_str[position:comma_position]
            position = comma_position + 1
            comma_position = self.csv_str.find(",", position)
        yield self.csv_str[position:]


for param in ListParam("un,deux,trois"):
    print(param)
